src/App.tsx

```diff
@@ -7,33 +8,49 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+export class App extends React.Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
 
-        {' time is '}
+  timerId = window.setInterval(() => {
```

Comment:
I think it's better to set the interval in componentDidMount() instead of directly defining timerId as a class property.